gulpfile.babel.js

```diff
@@ -116,41 +117,54 @@ gulp.task('server', function() {
 
 // Starts a garphene build that loads browser.html app from the localhost:6060
 gulp.task('gecko', function() {
-  var app = child.spawn(settings.geckoPath, [
-    '--profile',
-    settings.profilePath,
-    '--start-manifest=http://localhost:' + settings.port + '/manifest.webapp'
-  ], {
-    stdio: 'inherit'
+  fs.exists(settings.geckoPath, function (exists) {
+    if(exists) {
+      var app = child.spawn(settings.geckoPath, [
+          '--profile',
+          settings.profilePath,
+          '--start-manifest=http://localhost:' + settings.port + '/manifest.webapp'
+      ], {
+          stdio: 'inherit'
+      });
+      var exit = function(code) {
+          app.kill();
+          process.exit(code);
+      }
+
+      process.on('SIGINT', exit);
+      app.on('close', exit);
+    } else {
+      console.log("Error: Gecko binary not found: " + settings.geckoPath;);
+      process.exit(1);
+    } 
   });
-
-  var exit = function(code) {
-    app.kill();
-    process.exit(code);
-  }
-
-  process.on('SIGINT', exit);
-  app.on('close', exit);
 });
 
 gulp.task('servo', function() {
-  var app = child.spawn(settings.servoPath, [
-    '-w',
-    '-b',
-    '--pref',
-    'dom.mozbrowser.enabled',
-    'http://localhost:' + settings.port
-  ], {
-    stdio: 'inherit'
+  fs.exists(settings.geckoPath, function (exists) {
```

Comment:
Also, add a default path to `servoPath: process.env.BROWSER_HTML_SERVO_PATH`, like it is done for geckoPath. (`|| /usr/local/bin/servo`)